File: api/DTOs/Admin/Orders/OrderToCreateDto.cs
using System.ComponentModel.DataAnnotations;

namespace api.DTOs.Admin.Orders
{
    public class OrderToCreateDto
    {
        [Required]
        public DateTime OrderDate { get; set; }
        public string OrderRef { get; set; }
        public DateTime OrderRefDate {get; set;}
        [Required]
        public int CustomerId { get; set; }
        public string Country { get; set; }
        public string CityOfWorking { get; set; }
        public DateTime CompleteBy { get; set; }
        public string Remarks { get; set; }
        public ICollection<OrderItemDto> OrderItems { get; set; }
    }
}

File: client/src/app/_dtos/orders/orderToCreateDto.ts
import { IOrderItemDto } from "../admin/orderBriefDto";

export interface IOrderToCreateDto
{
    orderDate: Date;
    orderRef: string;
    orderRefDate: Date;
    customerId: number;
    country: string;
    cityOfWorking: string;
    completeBy: Date;
    remarks: string;
    orderItems: IOrderItemDto[];
}